planning/behavior_velocity_run_out_module/src/utils.cpp

```diff
@@ -176,6 +176,61 @@ std::optional<size_t> findFirstStopPointIdx(PathPointsWithLaneId & path_points)
   return {};
 }
 
+bool pathIntersectsEgoCutLine(
+  const std::vector<geometry_msgs::msg::Pose> & path, const geometry_msgs::msg::Pose & ego_pose,
+  const double half_line_length, std::vector<geometry_msgs::msg::Point> & ego_cut_lane)
+{
+  if (path.size() < 2) return false;
+  const auto ego_rpy = rpyFromQuat(ego_pose.orientation);
+  const double ego_yaw = ego_rpy.z;
+
+  auto getOffsetPoint = [](
+                          const geometry_msgs::msg::Pose & ego_pose, const double half_line_length,
+                          const double angle) {
+    geometry_msgs::msg::Point p;
+    p.x = ego_pose.position.x + half_line_length * std::cos(angle);
+    p.y = ego_pose.position.y + half_line_length * std::sin(angle);
+    return p;
+  };
+
+  const geometry_msgs::msg::Point p1 = getOffsetPoint(ego_pose, half_line_length, ego_yaw + M_PI_2);
+  const geometry_msgs::msg::Point p2 = getOffsetPoint(ego_pose, half_line_length, ego_yaw - M_PI_2);
+  ego_cut_lane = {p1, p2};
```

Comment:
You can use tier4_autoware_utils::calcOffsetPose(ref_pose, x, y, z). Then, you don't have to calculate yaw here.